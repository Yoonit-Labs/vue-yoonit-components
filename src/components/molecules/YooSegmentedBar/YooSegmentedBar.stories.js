import yooSegmentedBar from './YooSegmentedBar.vue'

export default {
  title: 'Yoonit/YooSegmentedBar',
  component: yooSegmentedBar,
  argTypes: {
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { yooSegmentedBar },
  template:
    '<yoo-segmented-bar v-bind="$props"></yoo-segmented-bar>'
})

export const Custom = Template.bind({})
Custom.args = { list: [{ title: 'First' }, { title: 'Second' }] }
