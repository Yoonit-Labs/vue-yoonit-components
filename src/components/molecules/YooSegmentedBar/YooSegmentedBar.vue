<template lang="pug">
  YooFlexLayout(
    flexDirection="row",
    alignContent="center",
    alignItems="center",
    justifyContent="center",
    flexWrap="nowrap",
    class="yoo-segmented-bar"
  )
    YooFlexLayout(
      v-for="(option, index) in list",
      :key="option.title",
      :class="index !== takeLastItem ? 'yoo-segmented-bar__item yoo-segmented-bar__item--border' : 'yoo-segmented-bar__item '"
      justifyContent="center"
    )
      YooButton(
        :text="option.title"
        variation="clear",
        :fill="index === selectedIndex ? 'primary' : 'default'",
        size="normal",
        style="width: 100%; height: 40px;"
        @onClick="onSelectedIndexChange(index)"
      )
</template>

<script>

import YooFlexLayout from '@/components/quarks/YooFlexLayout/YooFlexLayout.vue'

import YooButton from '@/components/atoms/YooButton/YooButton.vue'

export default {
  name: 'YooSegmentedBar',
  props: {
    list: {
      type: Array,
      required: true
    },
    currentIndex: {
      type: Number,
      default: 0
    }
  },
  components: {
    YooFlexLayout,
    YooButton
  },
  data: () => ({
    selectedIndex: 0
  }),
  mounted () {
    this.selectedIndex = this.currentIndex
  },
  computed: {
    takeLastItem () {
      return this.list.length - 1
    }
  },
  methods: {
    onSelectedIndexChange (index) {
      this.selectedIndex = index
      this.$emit('onClick', index)
    }
  }
}
</script>

<style src="./YooSegmentedBar.sass" lang="sass" scoped></style>
