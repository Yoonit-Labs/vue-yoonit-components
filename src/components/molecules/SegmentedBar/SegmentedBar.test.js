import { mount } from '@vue/test-utils'
import YooSegmentedBar from '@/components/molecules/SegmentedBar/SegmentedBar.vue'

const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return mount(YooSegmentedBar, {
    slots: { default: SlotText },
    propsData: {
      list: [
        { title: 'First' },
        { title: 'Second' }
      ]
    },
    data: () => ({
      selectedIndex: 1
    })
  })
}

describe('YooSegmentedBar Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', async () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', async () => {
    expect(wrapper.classes('yoo-segmented-bar')).toBe(true)
    expect(wrapper.find('.yoo-segmented-bar__item').exists()).toBe(true)
    expect(wrapper.find('button.yoo-btn').exists()).toBe(true)
  })

  describe('Events', () => {
    describe('onSelectedIndexChange', () => {
      it('Emits onSelectedIndexChange Event', async () => {
        wrapper.find('button.yoo-btn').trigger('click')
        expect(wrapper.emitted()).toHaveProperty('onClick')
      })
    })
  }) // describe Events
})
