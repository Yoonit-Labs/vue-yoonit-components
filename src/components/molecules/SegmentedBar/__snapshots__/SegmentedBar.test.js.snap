// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooSegmentedBar Component Matches Snapshot 1`] = `
<div class="yoo-flex-layout yoo-segmented-bar" style="flex-direction: row; flex-wrap: nowrap; justify-content: center; align-items: center; align-content: center;">
  <div class="yoo-flex-layout yoo-segmented-bar__item yoo-segmented-bar__item--border" style="flex-direction: row; flex-wrap: nowrap; justify-content: center; align-items: stretch; align-content: stretch;">
    <div class="yoo__container--button" size="normal" style="width: 100%; height: 40px;"><button class="yoo-btn yoo-btn__fill--primary yoo-btn--clear yoo-btn--normal yoo-btn--animation-none">
        <!---->
        <p class="yoo-btn__text yoo-btn__text--md">First</p>
        <!---->
      </button></div>
  </div>
  <div class="yoo-flex-layout yoo-segmented-bar__item " style="flex-direction: row; flex-wrap: nowrap; justify-content: center; align-items: stretch; align-content: stretch;">
    <div class="yoo__container--button" size="normal" style="width: 100%; height: 40px;"><button class="yoo-btn yoo-btn__fill--dark yoo-btn--clear yoo-btn--normal yoo-btn--animation-none">
        <!---->
        <p class="yoo-btn__text yoo-btn__text--md">Second</p>
        <!---->
      </button></div>
  </div>
</div>
`;
